src/main/java/uk/gov/hmcts/darts/arm/service/impl/CleanupArmResponseFilesServiceImpl.java

```diff
@@ -0,0 +1,230 @@
+                log.debug("Found ARM input upload file {} for cleanup", armInputUploadFilename);
+                if (armInputUploadFilename.endsWith(ArmResponseFilesHelper.generateSuffix(ARM_INPUT_UPLOAD_FILENAME_KEY))) {
+                    readInputUploadFile(externalObjectDirectory, armInputUploadFilename);
+                } else {
+                    log.warn("ARM response file {} not input upload file for EOD {}", armInputUploadFilename, externalObjectDirectory.getId());
+                }
+            }
+        } else {
+            log.info("Unable to find input upload file with prefix {} for cleanup with EOD {}", prefix, externalObjectDirectory.getId());
+        }
+    }
+
+    private void readInputUploadFile(ExternalObjectDirectoryEntity externalObjectDirectory, String armInputUploadFilename) {
+        try {
+            InputUploadFilenameProcessor inputUploadFilenameProcessor = new InputUploadFilenameProcessor(armInputUploadFilename);
+            String responseFilesHashcode = inputUploadFilenameProcessor.getHashcode();
+            log.debug("List response files starting with hashcode {}", responseFilesHashcode);
+            List<String> responseBlobs = armDataManagementApi.listResponseBlobs(responseFilesHashcode);
+            if (CollectionUtils.isNotEmpty(responseBlobs)) {
+                processResponseBlobs(responseBlobs, externalObjectDirectory, armInputUploadFilename);
+            } else {
+                log.warn("Unable to find responses files from Input Upload file {} for EOD {}", armInputUploadFilename, externalObjectDirectory.getId());
+            }
+        } catch (IllegalArgumentException e) {
+            // This occurs when the filename is not parsable
+            log.error("Unable to process filename {} for {} - {}", armInputUploadFilename, externalObjectDirectory.getId(), e.getMessage());
+        } catch (Exception e) {
+            log.error("Unable to list responses for file {} for {} - {}", armInputUploadFilename, externalObjectDirectory.getId(), e.getMessage());
+        }
+    }
+
+    private void processResponseBlobs(List<String> responseBlobs, ExternalObjectDirectoryEntity externalObjectDirectory,
+                                      String armInputUploadFilename) {
+        String createRecordFilename = null;
+        String uploadFilename = null;
+        String invalidFileFilename = null;
+
+        for (String responseFile : responseBlobs) {
+            if (responseFile.endsWith(ArmResponseFilesHelper.generateSuffix(ARM_CREATE_RECORD_FILENAME_KEY))) {
+                createRecordFilename = responseFile;
+            } else if (responseFile.endsWith(ArmResponseFilesHelper.generateSuffix(ARM_UPLOAD_FILE_FILENAME_KEY))) {
+                uploadFilename = responseFile;
+            } else if (responseFile.endsWith(ArmResponseFilesHelper.generateSuffix(ARM_INVALID_LINE_FILENAME_KEY))) {
+                invalidFileFilename = responseFile;
+            }
+        }
+
```

Comment:
maybe call armResponseFilesToBeDeleted.clear() here just to be safe - i can't tell whether it is or should be empty at this stage

Reply:
It is cleared at line 95 and 204 depending on which loop it is in